org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/Protocol.xtend

```diff
@@ -7053,9 +7053,12 @@ class FileOperationPattern {
 	/**
 	 * Whether to match files or folders with this pattern.
 	 *
+	 * The match kind could be <code>FileOperationPatternKind.File</code>,
+	 * <code>FileOperationPatternKind.Folder</code> and undefined.
+	 *
 	 * Matches both if undefined.
 	 */
-	FileOperationPatternKind matches
```

Comment:
is this a general new pattern how we deal with enums?

Reply:
should we also remove the enum then?

Reply:
Declaring it as a `String` is the standard pattern used in this project when the enum is a string. The *Kind class is just used as a helper in this case. It was simply a mistake to not make it a `String` initially.

Reply:
See #283

Reply:
I just quickly search the classes with `*Kind`, looks like we use enum for number type (such as `CompletionItemKind`), and string literal for string type (such as `FoldingRangeKind`).

`FileOperationPatternKind` accepts `file` and `folder` as value, so using string literals look ok.

Reply:
thank @KamasamaK for clarification.